rustls/src/atomic_sync.rs

```diff
@@ -0,0 +1,7 @@
+// NOTE: This `atomic_sync` module is intended to make it easier for a fork to
+// use another implementation of `Arc` such as `portable_atomic_util::Arc`,
+// which can help support build targets with no atomic pointer.
+// This module also makes it really easy for CI to overwrite this import
+// for build & unit testing with `portable_atomic_util::Arc`.
```

Comment:
I think we could improve this comment by describing more of how it makes it easier for these consumers. E.g. it's a level of indirection that gives a central place to change the `Arc` implementation. I think using Rustdoc tripple-slashes for the comment would be also be preferred.

```suggestion
/// The `atomic_sync` module indirects the `Arc` implementation to allow downstream forks
/// of rustls targetting architectures without atomics to replace the implementation in one
/// central location.
///
/// See `admin/use-portable-atomic-arc` and its use in CI for a demonstration.
```